src/libcore/iter/traits.rs

```diff
@@ -386,10 +386,11 @@ pub trait Extend<A> {
 /// ```
 #[stable(feature = "rust1", since = "1.0.0")]
 pub trait DoubleEndedIterator: Iterator {
-    /// An iterator able to yield elements from both ends.
+    /// Removes and returns an element from the end of the iterator.
```

Comment:
As I understand it, when I reread the array (but calling again `.iter()`), the elements have been removed. If that's the case then all good.